src/Services/BinaryWriterReaderExtensions.cs

```diff
@@ -1,6 +1,8 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+#if NETFRAMEWORK
```

Comment:
Since this and HostString are excluded entirely from non-netfx targets, I wonder if it might be cleaner to just exclude them at the csproj level. Maybe move both files under a 'framework' folder and exclude that folder from compilation for TFMs other than net472.

Reply:
moved back to the framework class only